src/combinatorics/simple_graph/prod.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2022 George Peter Banyard, Yaël Dillies, Kyle Miller. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: George Peter Banyard, Yaël Dillies, Kyle Miller
+-/
+import combinatorics.simple_graph.connectivity
+
+/-!
+# Graph products
+
+This file defines the box product of graphs and other product constructions. The box product of `G`
+and `H` is the graph on the product of the vertices such that `x` and `y` are related iff they agree
+on one component and the other one is related via either `G` or `H`. For example, the box product of
+two edges is a square.
+
+## Main declarations
+
+* `simple_graph.box_prod`: The box product.
+
+## Notation
+
+* `G □ H`: The box product of `G` and `H`.
+-/
+
+variables {α β γ : Type*}
+
+namespace simple_graph
+variables {G : simple_graph α} {H : simple_graph β} {I : simple_graph γ} {a a₁ a₂ : α} {b b₁ b₂ : β}
+  {x y : α × β}
+
+/--Box product of simple graphs. It relates `(a₁, b)` and `(a₂, b)` if `G` relates `a₁` and `a₂`,
+and `(a, b₁)` and `(a, b₂)` if `H` relates `b₁` and `b₂`. -/
+def box_prod (G : simple_graph α) (H : simple_graph β) : simple_graph (α × β) :=
```

Comment:
Are there any other obvious graphs of this signature? I assume `adj := λ x y, G.adj x.1 y.1 ∧ H.adj x.2 y.2,` is a valid graph; does it have a name you can mention so as to disambiguate?

Reply:
There are [at least 11 of them](https://en.wikipedia.org/wiki/Graph_product)! The ultimate goal of this file is to contain all the usual ones.

Reply:
I think that's worth adding as a reference and/or todo comment